linkml/utils/generator.py

```diff
@@ -190,6 +197,28 @@ def __post_init__(self) -> None:
             self.format = self.valid_formats[0]
         if self.format not in self.valid_formats:
             raise ValueError(f"Unrecognized format: {format}; known={self.valid_formats}")
+
+        # check if `deprecation_name` is in the global DEPRECATIONS list
+        # defined in linkml/utils/deprecation.py
+        if self.deprecation_name:
+            from linkml.utils.deprecation import DEPRECATIONS, deprecation_warning
+
+            # check if `deprecation_name` is in DEPRECATIONS
+            # log a warning message if we are using a `deprecation_name` other than
+            # anything defined in the global DEPRECATIONS list
+            if not any(d.name == self.deprecation_name for d in DEPRECATIONS):
+                self.logger.warning(f"Deprecation name '{self.deprecation_name}' not found in DEPRECATIONS list")
+
+            # ensure the `deprecated` flag is set if `deprecation_name` is provided
+            if not self.deprecated:
+                self.logger.warning(
+                    f"Setting deprecated=True for generator with deprecation_name='{self.deprecation_name}'"
+                )
+                self.deprecated = True
+
+            # issue the deprecation warning
+            deprecation_warning(self.deprecation_name)
+
```

Comment:
I'm not sure why we would want to do this rather than just emitting the `deprecation_warning` in the post init?

Reply:
we would also probably want tests for this

Reply:
Makes sense! i'll add a test for this. I'll add a test to `tests/test_generators/test_markdowngen.py`.